Add check pull request reviews action

adcreare opened the pull request:
Closes https://github.com/checkdigit/hash/issues/15
Added two reviewers. 
Action should run after the first review and then fail until the second reviewer approves..

carlansley commented:
@adcreare it worked, but I had to make that check "required" in the branch protection rules

carlansley commented:
Actually I don't think it worked.. it failed with this error which I assume isn't right:
`Run checkdigit/github-actions/check-pr-reviews@prepare-beta
Action Error - exit 1 - error: Error: incorrect action configuration
    at haveAllReviewersReviewed (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/github-api/index.js:96:15)
    at reviewedCorrectly (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/github-api/index.js:164:40)
    at main (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/check-pr-reviews/index.js:15:61)
    at Object.<anonymous> (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/check-pr-reviews/index.js:21:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47`

adcreare commented:
> @adcreare it worked, but I had to make that check "required" in the branch protection rules

Its actually failed somewhere (I mean it was supposed to fail, but it failed with incorrect action configuration) - thankfully i can run it again so I'll sort that out.

carlansley commented:
@adcreare maybe it needs to make a comment (e.g. 1/2 reviewers approved).  or some way of providing feedback

adcreare commented:
For some reason GITHUB_TOKEN is not being passed to the action. I think this might be because I'm using the `pull_request_review` hook vs the more normal ones. 
I've updated the action configuration here to push that environment var and I've added additional logging on the action - maybe give it another review and see what happens. 

Once that is sorted out I'll add in the posting of a feedback about number of reviewers left.

carlansley commented:
the logs look good? but didn't bomb the check

adcreare commented:
Urgh - https://github.com/checkdigit/hash/actions/runs/5183818967 even has the annotation that the check did not pass correctly, but seems thats not enough for the action to fail in the view here. 
I'm going to have to convert that to an exception being thrown i guess instead of setError().

adcreare commented:
Ok issued sorted out and comments added it. I'll wait for Luke to review to make sure the logic for all reviewers have approved works correctly.

lukebrdn reviewed (dismissed):
lgtm

adcreare commented:
> ✅ PR review status - All reviews completed and approved!

^ Just for you @lukebrdn  :)

adcreare commented:
Ready for another look - just moved to main on the actions now the github-actions pr has been merged

carlansley reviewed (changes requested):
Just realized the ci.yml doesn't actually execute any tests

carlansley commented on .github/workflows/publish-beta.yml:
Might be worth adding the comment action here?

adcreare commented:
Done - ready for another look

lukebrdn reviewed (approved):
lgtm
